app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt

```diff
@@ -0,0 +1,63 @@
+package au.com.gridstone.trainingkotlin
+
+import android.app.Application
+import au.com.gridstone.trainingkotlin.BuildConfig.BASAE_URL
+import au.com.gridstone.trainingkotlin.apiservices.PokemonService
+import au.com.gridstone.trainingkotlin.viewmodels.DetailViewModel
+import au.com.gridstone.trainingkotlin.viewmodels.HomeViewModel
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.OkHttpClient
+import org.koin.android.ext.koin.androidContext
+import org.koin.core.context.GlobalContext.startKoin
+import org.koin.core.module.Module
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+const val POKEMON: String = "pokemon"
+const val SCOPE_DETAIL: String = "detailScope"
+
+val applicationModule: Module = module {
+  single<Retrofit> {
+    Retrofit.Builder()
+      // .baseUrl(Constants.BASE_URL)
+      .baseUrl(BASAE_URL)
+      .client(get())
+      // Moshi maps JSON to classes
+      .addConverterFactory(MoshiConverterFactory.create(
+        Moshi.Builder()
+          .add(KotlinJsonAdapterFactory())
+          .build()))
+      .build()
+  }
+
+  single {
+    OkHttpClient()
+      .newBuilder()
+      .addInterceptor(LoggingInterceptor)
+      .build()
+  }
+
+  single<PokemonService> {
+    get<Retrofit>().create(PokemonService::class.java)
+  }
+
+  single { HomeViewModel() }
+
+  scope(named(SCOPE_DETAIL)) {
+    scoped { (id: String) -> DetailViewModel(id) }
+  }
```

Comment:
is this intended? 🤔

Reply:
sorry was not my intention to commit this I will be aware next time thanks for pointing this out.